File: backends/python-fastapi/app/domain/staff/value_objects.py
"""
スタッフドメイン バリューオブジェクトモジュール。

Author: Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
"""
from dataclasses import dataclass
from datetime import datetime
from typing import Optional


@dataclass(frozen=True)
class StaffListItem:
    """スタッフ一覧レスポンス用バリューオブジェクト。"""

    id: int
    name: str
    email: str
    role: int
    status: int  # 0=削除済み, 1=有効
    created_at: Optional[datetime]
    updated_at: Optional[datetime]


@dataclass(frozen=True)
class StaffVo:
    """スタッフ詳細レスポンス用バリューオブジェクト。"""

    id: int
    name: str
    avatar: Optional[str]
    role: int

File: backends/ts-hono/src/domain/staff/valueObjects.ts
/**
 * スタッフドメイン バリューオブジェクトモジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */

/** スタッフ一覧レスポンス用 VO。 */
export interface StaffListItem {
  id: number;
  name: string;
  email: string;
  role: number;
  /** 0=削除済み, 1=有効 */
  status: number;
  createdAt: Date | null;
  updatedAt: Date | null;
}

/** ログイン中スタッフ情報レスポンス用 VO。 */
export interface StaffVo {
  id: number;
  name: string;
  avatar: string | null;
  role: number;
}

/** スタッフ upsert 入力 VO（リポジトリ用）。 */
export interface StaffUpsertVo {
  provider: number;
  providerId: string;
  name: string;
  email: string;
  avatar?: string;
  role?: number;
}
